animated-transformer/src/lib/transformer/dropout.ts

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2023 Google LLC. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {
+    GTensor,
+  } from '../gtensor/gtensor';
+
+  import {dropout as tf_dropout} from '@tensorflow/tfjs';
+  
+  // Wrapper for tf ts dropout.
+  export function dropout<G extends string, D extends G>(
+    dropoutRate: number,
+    g: GTensor<G>,
+    seed?: number,
+    dimNames?: string[],
```

Comment:
the type of dimNames should be G[] (or D) I think, since is the set of dimension names (this will catch you accidentally trying to dropout on a non-existent dimension). I think in this case, adding the D extends G doesn't actually help you, so directly using G[] for dimNames will be fine.

Reply:
Done